### PR TITLE
Code sanitising: type checks

### LINKED ISSUE #4834: Update input checks to use rlang type checking
This is an issue that is mostly about code sanitising. The code base is littered with checks for e.g. various types of scalars etc for which rlang now provides good self-documenting checks.

### PR DESCRIPTION
**Pull request by teunbrand**
This PR aims to fix #4834.

It replaces checks of the type `is.myclass(x) && length(x) == 1` by {rlang} scalar checks. There was no function for scalar numeric checks, so I added one.

### CODE REVIEW

**Comment by teunbrand** [comment]
TODO: use_standalone("r-lib/rlang", "types-check") instead of home grown argument checkers

**Comment by teunbrand** [comment]
I'm putting this PR on hold pending the release of rlang 1.1.0, which is required for the imported functions.

File: R/facet-wrap.R [hunk 1]
```diff
@@ -99,12 +99,17 @@ facet_wrap <- function(facets, nrow = NULL, ncol = NULL, scales = "fixed",
...
+
+  check_number_whole(ncol, allow_null = TRUE, min = 1)
+  check_number_whole(nrow, allow_null = TRUE, min = 1)
+
+  # if (!(is.null(ncol) || (is_integerish(ncol, 1) && ncol > 0))) {
+  #   cli::cli_abort("{.arg ncol} must be a positive scalar integer or {.val NULL}")
+  # }
+  # if (!(is.null(nrow) || (is_integerish(nrow, 1) && nrow > 0))) {
+  #   cli::cli_abort("{.arg nrow} must be a positive scalar integer or {.val NULL}")
+  # }
```
Comment (teunbrand) [inline R/facet-wrap.R:111]: Currently gives misleading message.
I've filed https://github.com/r-lib/rlang/issues/1591 in the hope to improve the message a bit.
If we can't improve the message, we'll revert back to the old check.

File: tests/testthat/test-prohibited-functions.R [hunk 2]
```diff
@@ -68,7 +68,8 @@ test_that("do not use stop()", {
 })
 
 test_that("do not use abort()", {
-  aborts <- vapply(R_files, get_n_abort, integer(1))
+  files <- R_files[!grepl("^import-standalone-", basename(R_files))]
+  aborts <- vapply(files, get_n_abort, integer(1))
```
Comment (teunbrand) [inline tests/testthat/test-prohibited-functions.R:72]: Is it OK to ignore these?

Comment (thomasp85) [reply tests/testthat/test-prohibited-functions.R:72]: yes, that is needed

File: R/facet-wrap.R (hunk 1 above)
Comment (teunbrand) [reply R/facet-wrap.R]: Fixed by https://github.com/r-lib/rlang/pull/1594

**Comment by teunbrand** [comment]
I think this PR is now ready for review, but should wait until rlang 1.1.0 has landed on CRAN

**Review by thomasp85 — APPROVED** [review]
LGTM - have you checked that all checks are covered by unit tests (they should as I went through this when updating the errors before last minor release but better check again

**Comment by thomasp85** [comment]
hopefully rlang will be released this week

**Comment by teunbrand** [comment]
> have you checked that all checks are covered by unit tests

I'll double-check this later today

**Comment by thomasp85** [comment]
rlang 1.1.0 is now on CRAN

**Pull request merged**